nimare/utils.py

```diff
@@ -478,6 +482,99 @@ def memmap_context(self, *args, **kwargs):
     return inner_function
 
 
+BYTE = 2
+KILOBYTE = BYTE ** 10
+BYTE_CONVERSION = {
+    "kb": KILOBYTE,
+    "mb": KILOBYTE ** 2,
+    "gb": KILOBYTE ** 3,
+    "tb": KILOBYTE ** 4,
+}
+
+
+def determine_chunk_size(limit, arr, multiplier=1):
+    """Determine how many arrays can be read into memory at once.
+
+    Parameters
+    ----------
+    limit : :obj:`str`
+        String representation of memory limit, can use:
+        kb, mb, gb, and tb as suffix (e.g., "4gb").
+    arr : :obj:`numpy.array`
+        Representative numpy array.
+    multiplier : :obj:`int`
+        Adjustment for processes that have more or
+        less overhead than expected.
+    """
+    limit = limit.lower()
+    size, representation = re.search(r"([0-9]+)([a-z]+)", limit).groups()
```

Comment:
It might be nice to validate the string here so that exceptions will be maximally informative.